### PR TITLE
test(ticdc): migrate test-infra to testify for cdc/kv pkg (#2899)

### LINKED ISSUE #2899: migrate test-infra to testify for `cdc/kv` pkg

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #2899

### What is changed and how it works?
Migrate all test cases under cdc/kv to testify.

### Check List 

Tests 

 - Unit test

### Release note 

```release-note
`None`
```

### CODE REVIEW

**Review by overvenus — APPROVED** [review]

File: cdc/kv/token_region_test.go [hunk 1]
```diff
@@ -43,31 +41,33 @@ func (s *tokenRegionSuite) TestRouter(c *check.C) {
...
+	defer testleak.AfterTestT(t)
+	t.Parallel()
+	testRouterWithConsumer(t, func() {})
 }
 
-func (s *tokenRegionSuite) TestRouterWithSlowConsumer(c *check.C) {
-	defer testleak.AfterTest(c)()
-	s.testRouterWithConsumer(c, func() { time.Sleep(time.Millisecond * 15) })
+func TestRouterWithSlowConsumer(t *testing.T) {
+	defer testleak.AfterTestT(t)
```
Comment (asddongmen) [inline cdc/kv/token_region_test.go:65]: Why not remove this line: `defer testleak.AfterTestT(t)`?

Comment (CharlesCheung96) [reply cdc/kv/token_region_test.go]: Already removed, PTAL

File: cdc/kv/client_test.go [hunk 2]
```diff
@@ -1499,24 +1456,22 @@ eventLoop:
...
 
 // TestStreamRecvWithErrorAndResolvedGoBack mainly tests the scenario that the `Recv` call of a gPRC
 // stream in kv client meets error, and kv client reconnection with tikv with the current tso
-func (s *clientSuite) TestStreamRecvWithErrorAndResolvedGoBack(c *check.C) {
-	defer testleak.AfterTest(c)()
-	defer s.TearDownTest(c)
+func TestStreamRecvWithErrorAndResolvedGoBack(t *testing.T) {
 	if !util.FailpointBuild {
-		c.Skip("skip when this is not a failpoint build")
+		t.Skip("skip when this is not a failpoint build")
```
Comment (CharlesCheung96) [inline cdc/kv/client_test.go:1467]: We need to confirm that the substitutions here are equivalent

**Review by ben1009 — APPROVED** [review]

**Pull request merged**